build/parseSpec.c

```diff
@@ -501,17 +506,24 @@ int readLine(rpmSpec spec, int strip)
 	goto retry;
     }
 
-after_classification:
-    if (match != -1) {
+    if (lineType->id & (LINE_IF | LINE_IFARCH | LINE_IFNARCH | LINE_IFOS |
+	    LINE_IFNOS)) {
```

Comment:
This calls for a bitmask, something like LINE_IFANY, which can be used to trap all the types with one symbol. Makes the line shorter and more readable. (fwiw no I'm not reviewing, just taking a look)